lib/balanced-pool.js

```diff
@@ -39,8 +55,33 @@ class BalancedPool extends PoolBase {
     ))) {
       return this
     }
+    const pool = new Pool(upstream, Object.assign({}, this[kOptions]))
+
+    this[kAddClient](pool)
+
+    pool.on('connectionError', () => {
+      pool[kWeight] = Math.max(1, pool[kWeight] - 10)
+
+      this._updateBalancedPoolStats()
+    })
+
+    pool.on('disconnect', (...args) => {
+      const err = args[2]
+      if (err) {
```

Comment:
```suggestion
      if (err && err.code !== 'UND_ERR_INFO') {
```

Reply:
you just helped me found a weird behaviour I had in the tests :D